metrics-core/src/main/java/com/codahale/metrics/CsvReporter.java

```diff
@@ -239,21 +241,32 @@ private void report(long timestamp, String name, String header, String line, Obj
             final File file = new File(directory, sanitize(name) + ".csv");
             final boolean fileAlreadyExists = file.exists();
             if (fileAlreadyExists || file.createNewFile()) {
-                final PrintWriter out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(file,true), UTF_8));
-                try {
-                    if (!fileAlreadyExists) {
-                        out.println("t," + header);
-                    }
-                    out.printf(locale, String.format(locale, "%d,%s%n", timestamp, line), values);
-                } finally {
-                    out.close();
+                final PrintWriter out;
+                if (printWriterHashMap.containsKey(file)) {
+                    out = printWriterHashMap.get(file);
+                } else {
+                    out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(file,true), UTF_8));
+                    printWriterHashMap.put(file, out);
                 }
+
+                if (!fileAlreadyExists) {
+                    out.println("t," + header);
+                }
+                out.printf(locale, String.format(locale, "%d,%s%n", timestamp, line), values);
             }
         } catch (IOException e) {
             LOGGER.warn("Error writing to {}", name, e);
         }
     }
 
+    @Override
+    public void stop() {
+        for (final PrintWriter pw : printWriterHashMap.values()) {
```

Comment:
I am not sure if it is ensured that each metric is removed on the `Reporter` side.